client/navigation/components/intro-modal/test/index.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * External dependencies
+ */
+import { fireEvent, render, screen } from '@testing-library/react';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { IntroModal, INTRO_MODAL_DISMISSED_OPTION_NAME } from '../';
+
+global.window.wcNavigation = {};
+
+jest.mock( '@wordpress/data', () => {
+	// Require the original module to not be mocked...
+	const originalModule = jest.requireActual( '@wordpress/data' );
+
+	return {
+		__esModule: true, // Use it when dealing with esModules
+		...originalModule,
+		useDispatch: jest.fn().mockReturnValue( {} ),
+		useSelect: jest.fn().mockReturnValue( {} ),
+	};
+} );
+
+describe( 'IntroModal', () => {
+	test( 'should not show when modal options are resolving', () => {
+		useSelect.mockImplementation( () => ( {
+			isWelcomeModalShown: null,
+			isResolving: true,
+		} ) );
+
+		const { container } = render( <IntroModal /> );
+
+		expect( container ).toBeEmptyDOMElement();
+	} );
+
+	test( 'should dismiss when the welcome modal is shown', () => {
+		const updateOptions = jest.fn();
+		useSelect.mockImplementation( () => ( {
+			isWelcomeModalShown: true,
+		} ) );
+		useDispatch.mockImplementation( () => ( {
+			updateOptions,
+		} ) );
+
+		const { container } = render( <IntroModal /> );
+
+		expect( container ).toBeEmptyDOMElement();
+		expect( updateOptions ).toHaveBeenCalledWith( {
+			[ INTRO_MODAL_DISMISSED_OPTION_NAME ]: 'yes',
+		} );
```

Comment:
If the modal has already been shown, `dismissModal` is called. Am I understanding correctly that `dismissModal` is called on every screen, on every page refresh? If so, that function calls not only `updateOptions` but `recordEvent` as well. Is it worth revisiting this logic?

Reply:
The dismissal is only called when the welcome modal is shown simultaneously and only on the homescreen where the components have been mounted.

I think it's pretty unlikely that it gets called multiple times, but just in case, I added https://github.com/woocommerce/woocommerce-admin/pull/6518/commits/df134316909f500072746ffa33b034af7024b5b7 as a backup measure.

Reply:
Ok gotcha, thanks